ci/L0_accuracy_test/vllm_accuracy_test.py

```diff
@@ -0,0 +1,122 @@
+from tritonclient.utils import *
+from vllm import SamplingParams
+from vllm.engine.arg_utils import AsyncEngineArgs
+from vllm.engine.async_llm_engine import AsyncLLMEngine
+from vllm.utils import random_uuid
+
+sys.path.append("../common")
+from test_util import TestResultCollector, UserData, callback, create_vllm_request
+
+
+async def generate_python_vllm_output(prompt, llm_engine):
+    request_id = random_uuid()
+    sampling_parameters = {"temperature": 0.1, "top_p": 0.95}
+    sampling_params = SamplingParams(**sampling_parameters)
+
+    python_vllm_output = None
+    last_output = None
+
+    async for vllm_output in llm_engine.generate(prompt, sampling_params, request_id):
+        last_output = vllm_output
+
+    if last_output:
+        python_vllm_output = [
+            (prompt + output.text).encode("utf-8") for output in last_output.outputs
+        ]
+
+    return python_vllm_output
+
+
+class VLLMTritonAccuracyTest(TestResultCollector):
+    def setUp(self):
+        self.triton_client = grpcclient.InferenceServerClient(url="localhost:8001")
+        vllm_engine_config = {
+            "model": "facebook/opt-125m",
+            "gpu_memory_utilization": 0.3,
+        }
+
+        self.llm_engine = AsyncLLMEngine.from_engine_args(
+            AsyncEngineArgs(**vllm_engine_config)
+        )
+        self.vllm_model_name = "vllm_opt"
+
+    def test_vllm_model(self):
+        user_data = UserData()
+        stream = False
+        prompts = [
+            "The most dangerous animal is",
+            "The capital of France is",
+            "The future of AI is",
+        ]
+        number_of_vllm_reqs = len(prompts)
+        sampling_parameters = {"temperature": "0.1", "top_p": "0.95"}
+        python_vllm_output = []
+        triton_vllm_output = []
+
+        self.triton_client.start_stream(callback=partial(callback, user_data))
+        for i in range(number_of_vllm_reqs):
+            request_data = create_vllm_request(
+                prompts[i], i, stream, sampling_parameters, self.vllm_model_name
+            )
+            self.triton_client.async_stream_infer(
+                model_name=self.vllm_model_name,
+                request_id=request_data["request_id"],
+                inputs=request_data["inputs"],
+                outputs=request_data["outputs"],
+                parameters=sampling_parameters,
+            )
+
+            python_vllm_output.extend(
+                asyncio.run(generate_python_vllm_output(prompts[i], self.llm_engine))
+            )
+
+        for i in range(number_of_vllm_reqs):
+            result = user_data._completed_requests.get()
+            self.assertIsNot(type(result), InferenceServerException)
+
+            output = result.as_numpy("text_output")
+            self.assertIsNotNone(output)
+
+            triton_vllm_output.extend(output)
+
+        self.triton_client.stop_stream()
+        self.assertEqual(python_vllm_output, triton_vllm_output)
```

Comment:
@pskiran1 Will this ever fail? I think vLLM has different outputs for the same prompt sometimes, and I don't see a seed here to make this deterministic.

Reply:
During the testing, performed 100 iterations and observed no failures.
However, noticed that when the temperature was set to 0.5, it resulted in a failure.
To eliminate randomness and achieve consistent LLM output, adjusted the temperature to 0 and top_p to 1.
Conducted the test for 100 iterations again to ensure that there was no failure.

Reply:
Great work!